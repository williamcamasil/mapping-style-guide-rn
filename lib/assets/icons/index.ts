export * as Large from './large';
export * as Heavy from './heavy';
export * as Thin from './thin';
export * as Small from './small';
export * as Default from './default';
