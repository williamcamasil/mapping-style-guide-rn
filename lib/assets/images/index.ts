export { default as BtnTakePicture } from './BtnTakePicture';
export { default as BulletPoint } from './BulletPoint';
export { default as CheckboxImg } from './CheckboxImg';
export { default as DialogClose } from './DialogClose/index.svg';
export { default as Lock } from './Lock';
export { default as MaskedCardNumber } from './MaskedCardNumber';
export { default as Pointer } from './Pointer';
export { default as RadioImg } from './RadioImg';
