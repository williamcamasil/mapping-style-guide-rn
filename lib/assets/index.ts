export * as Icons from './icons';
export * as Images from './images';
export * as Lotties from './lotties';
