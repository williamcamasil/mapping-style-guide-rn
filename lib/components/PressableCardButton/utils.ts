import { AppThemeType } from '../../theme';

export const getButtonColor = (
  pressed?: boolean,
) => {
  if (pressed) return 'feedbackSuccess500';

  return 'primaryMain';
};

export const getIconColor = (
  theme: AppThemeType,
  pressed?: boolean,
) => {
  if (pressed) return theme.colors.feedbackSuccess500;

  return theme.colors.primaryMain;
};
