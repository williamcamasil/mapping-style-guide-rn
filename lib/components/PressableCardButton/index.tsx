import React, { ComponentType, useCallback, useState } from 'react';

import { SvgProps } from 'react-native-svg';

import { Icons } from '../../assets';
import { useViewStyles } from '../../hooks/useStyles';
import { withTheme } from '../../theme';
import { CardPropsType } from '../Card';
import CardIcon from '../CardIcon';
import Text from '../Text';
import { getButtonColor, getIconColor } from './utils';

export type PressableCardButtonPropsType<IconProps extends SvgProps = SvgProps> = CardPropsType & {
  title: string;
  pressedTitle: string;
  Icon?: ComponentType<IconProps>;
  PressedIcon?: ComponentType<IconProps>;
  onPress: () => void;
  testID?: string;
};

const PressableCardButton: React.FC<PressableCardButtonPropsType> = ({
  title,
  pressedTitle,
  Icon,
  PressedIcon,
  onPress,
  testID,
  theme,
  ...others
}) => {
  const [pressed, setPressed] = useState<boolean>(false);

  const buttonStyle = useViewStyles(() => [
    {
      backgroundColor: pressed ? theme.colors.feedbackSuccess100 : theme.colors.primary100,
    },
  ], [pressed, theme.colors.feedbackSuccess100, theme.colors.primary100]);

  const buttonColor = getButtonColor(pressed);
  const iconColor = getIconColor(theme, pressed);

  const handlePress = useCallback(() => {
    onPress();
    setPressed(true);
  }, [onPress]);

  return (
    <CardIcon
      {...others}
      onPress={handlePress}
      testID={testID}
      size="small"
      Icon={pressed ? PressedIcon : Icon}
      iconColor={iconColor}
      showShadow={false}
      style={buttonStyle}
      borderColor="transparent"
    >
      <Text variant="small" color={buttonColor}>
        {pressed ? pressedTitle : title}
      </Text>
    </CardIcon>
  );
};

PressableCardButton.defaultProps = {
  PressedIcon: Icons.Default.Success,
};

export default withTheme(PressableCardButton);
