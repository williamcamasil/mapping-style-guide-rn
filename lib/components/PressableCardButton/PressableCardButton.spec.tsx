import 'react-native';
import React from 'react';

import {
  render, fireEvent,
} from '@testing-library/react-native';

import PressableCardButton from '.';
import { Icons } from '../../assets';

describe('PressableCardButton Component', () => {
  it('should render button with text Copiar código de barras and press button and text change to Código copiado', () => {
    const mockFunction = jest.fn();
    const result = render(
      <PressableCardButton
        title={'Copiar código\nde barras'}
        pressedTitle={'Código\ncopiado'}
        Icon={Icons.Default.Copy}
        PressedIcon={Icons.Default.Success}
        onPress={mockFunction}
        testID="copy-button"
      />,
    );

    expect(result.getByText('Copiar código\nde barras')).toBeDefined();

    const copyButton = result.getByTestId('copy-button');
    fireEvent.press(copyButton);

    expect(mockFunction).toHaveBeenCalled();
    expect(result.getByText('Código\ncopiado')).toBeDefined();
  });

  it('should render button with text Baixar boleto and press button and text change to Boleto baixado', () => {
    const mockFunction = jest.fn();
    const result = render(
      <PressableCardButton
        title={'Baixar\nboleto'}
        pressedTitle={'Boleto\nbaixado'}
        Icon={Icons.Default.Copy}
        PressedIcon={Icons.Default.Success}
        onPress={mockFunction}
        testID="download-button"
      />,
    );

    expect(result.getByText('Baixar\nboleto')).toBeDefined();

    const copyButton = result.getByTestId('download-button');
    fireEvent.press(copyButton);

    expect(mockFunction).toHaveBeenCalled();
    expect(result.getByText('Boleto\nbaixado')).toBeDefined();
  });
});

describe('PressableCardButton snapshots', () => {
  it('default copy button', () => {
    const mockFunction = jest.fn();
    const result = render(
      <PressableCardButton
        title={'Copiar código\nde barras'}
        pressedTitle={'Código\ncopiado'}
        Icon={Icons.Default.Copy}
        PressedIcon={Icons.Default.Success}
        onPress={mockFunction}
      />,
    ).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('default download button', () => {
    const mockFunction = jest.fn();
    const result = render(
      <PressableCardButton
        title={'Baixar\nboleto'}
        pressedTitle={'Boleto\nbaixado'}
        Icon={Icons.Default.Copy}
        PressedIcon={Icons.Default.Success}
        onPress={mockFunction}
      />,
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});
