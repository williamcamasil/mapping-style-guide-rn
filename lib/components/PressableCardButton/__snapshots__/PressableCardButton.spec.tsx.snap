// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressableCardButton snapshots default copy button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#EDF1F8",
      "borderColor": "transparent",
      "borderRadius": 12,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 120,
      "opacity": 1,
      "padding": 16,
      "width": 135,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "space-between",
        }
      }
      testID="card-icon-view-content"
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          fill="none"
          focusable={false}
          height={24}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "fontFamily": "Averta CY",
                "fontSize": 13,
                "fontWeight": "bold",
                "marginRight": 12,
              },
              {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup
            fill={null}
            font={
              {
                "fontFamily": "Averta CY",
                "fontSize": "13",
                "fontWeight": "bold",
              }
            }
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M5 16h-.556A1.445 1.445 0 0 1 3 14.556V4.444C3 3.647 3.647 3 4.444 3h10.111C15.353 3 16 3.647 16 4.444V5m3.556 16H9.444A1.445 1.445 0 0 1 8 19.556V9.444C8 8.647 8.647 8 9.444 8h10.111C20.353 8 21 8.647 21 9.444v10.111c0 .798-.647 1.445-1.444 1.445Z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4278268296,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="1.8"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#013188",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 14,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 17,
          }
        }
      >
        Copiar c√≥digo
de barras
      </Text>
    </View>
  </View>
</View>
`;

exports[`PressableCardButton snapshots default download button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#EDF1F8",
      "borderColor": "transparent",
      "borderRadius": 12,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 120,
      "opacity": 1,
      "padding": 16,
      "width": 135,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "space-between",
        }
      }
      testID="card-icon-view-content"
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          fill="none"
          focusable={false}
          height={24}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "fontFamily": "Averta CY",
                "fontSize": 13,
                "fontWeight": "bold",
                "marginRight": 12,
              },
              {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup
            fill={null}
            font={
              {
                "fontFamily": "Averta CY",
                "fontSize": "13",
                "fontWeight": "bold",
              }
            }
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M5 16h-.556A1.445 1.445 0 0 1 3 14.556V4.444C3 3.647 3.647 3 4.444 3h10.111C15.353 3 16 3.647 16 4.444V5m3.556 16H9.444A1.445 1.445 0 0 1 8 19.556V9.444C8 8.647 8.647 8 9.444 8h10.111C20.353 8 21 8.647 21 9.444v10.111c0 .798-.647 1.445-1.444 1.445Z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4278268296,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="1.8"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#013188",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 14,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 17,
          }
        }
      >
        Baixar
boleto
      </Text>
    </View>
  </View>
</View>
`;
